src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,63 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, container, label, widget, enabled):
+        self._container = container
+        self._index = len(container)
```

Comment:
This should be a property of assignment, not construction. If you were to insert a tab at position N, "index" will only be correct if N is the last position in the list.